astroid/interpreter/_import/util.py

```diff
@@ -18,18 +22,46 @@ def is_namespace(modname: str) -> bool:
     # That's unacceptable here, so we fallback to _find_spec_from_path(), which does
     # not, but requires instead that each single parent ('astroid', 'nodes', etc.)
     # be specced from left to right.
-    components = modname.split(".")
-    for i in range(1, len(components) + 1):
-        working_modname = ".".join(components[:i])
+    processed_components = []
+    last_submodule_search_locations: _NamespacePath | None = None
+    for component in modname.split("."):
+        processed_components.append(component)
+        working_modname = ".".join(processed_components)
         try:
-            # Search under the highest package name
-            # Only relevant if package not already on sys.path
-            # See https://github.com/python/cpython/issues/89754 for reasoning
-            # Otherwise can raise bare KeyError: https://github.com/python/cpython/issues/93334
-            found_spec = _find_spec_from_path(working_modname, components[0])
+            # the path search is not recursive, so that's why we are iterating
```

Comment:
Isn't this duplicate of L23-24?

Reply:
Ah, sort of. Yeah. Both the modname and the path need to be iteratively built, with the `path` lagging by one. I was trying to explain why #1575 probably should not have just taken the top-most parent for `path`.